File: Front/src/type/Story.ts
import { Task } from "./Task";

export type Story = {
    id: string,
    name: string,
    description: string,
    creationDate: string,
    priority: StoryPriority,
    state: StoryState,
    tasks: Task[],
}

export enum StoryPriority
{
    Low,
    Medium,
    High
}

export enum StoryState
{
    Todo,
    Doing,
    Done
}

File: Back/Entities/Story.cs
using Back.Entities.Enums;

namespace Back.Entities;

public class Story
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    
    public string Description { get; set; }
    
    public Priority Priority { get; set; }
    
    public int ProjectId { get; set; }
    
    public DateTime CreationDate { get; set; }
    
    public State State { get; set; }
    
    public ICollection<StoryTask> Tasks { get; set; }
    
    public User User { get; set; }
    
    public int UserId { get; set; }
}